### PR TITLE
Add common definition files

### PR DESCRIPTION
**Pull request by anthony-c-martin**

### CODE REVIEW

File: schemas/common/definitions.json [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the copy"
+        },
+        "count": {
+          "oneOf": [
+            {
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
```
Comment (Tiano2017) [inline schemas/common/definitions.json]: >https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression [](start = 23, length = 91)

use relative path?

File: schemas/common/definitions.json [hunk 2]
```diff
@@ -0,0 +1,257 @@
+{
+  "id": "https://schema.management.azure.com/schemas/common/definitions.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "definitions",
+  "description": "Common definitions",
+  "definitions": {
```
Comment (Tiano2017) [inline schemas/common/definitions.json]: >definitions [](start = 3, length = 11)

we have something in deploymentTemplateParameters.json. Do we need to copy that over?

Comment (anthony-c-martin) [reply schemas/common/definitions.json]: No - that was a deliberate omission. All of the definitions I could find in deploymentTemplateParameters.json were specific to the api-version.

File: schemas/common/definitions.json (hunk 1 above)
Comment (anthony-c-martin) [reply schemas/common/definitions.json]: Yup, let me fix that.

File: schemas/common/autogeneratedResources.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "id": "https://schema.management.azure.com/schemas/common/autogeneratedResources.json#",
```
Comment (majastrz) [inline schemas/common/autogeneratedResources.json]: >id [](start = 3, length = 2)

Do we actually want to call the new schema "autogenerated"? From the customer POV, won't that sound a little weird?

File: schemas/common/manuallyAddedResources.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "id": "https://schema.management.azure.com/schemas/common/manuallyAddedResources.json#",
```
Comment (majastrz) [inline schemas/common/manuallyAddedResources.json]: >id [](start = 3, length = 2)

Same comment about naming here as well.

**Pull request merged**